go/src/github.com/RedHatInsights/insights-ocp-controller/pkg/controller/controller.go

```diff
@@ -99,6 +99,13 @@ func (c *Controller) ScanImages() {
 func (c *Controller) scanImage(id string, args []string, imageRef string, imageSha string) error {
 	endpoint := "unix:///var/run/docker.sock"
 	client, err := docker.NewVersionedClient(endpoint, "1.22")
+
+	_, err2 := client.InspectImage(id)
+	if err2 != nil {
+		log.Printf("Error testing if image %s exists: %s\n", id, err2)
```

Comment:
@gravitypriest take a look at the canScan implementation by @brantleyr (https://github.com/RedHatInsights/insights-ocp-controller/pull/6 )
 I think we should make this check  before we  contact the Master to request scan permission, to avoid asking for permission to scan before we know we can -  otherwise we will need to go back to the master to deque .